app/services/umrdr/date_range_service.rb

```diff
@@ -0,0 +1,73 @@
+module Umrdr
+  class DateRangeService  
+
+    attr_reader :params
+    
+    def initialize(params)
+      @params = params
+    end  
+
+    def transform
+
+      if (params[:date_coverage_1_year].length>0)
+        yyyy1= params[:date_coverage_1_year].to_i 
+      else
+        yyyy2 = 0 #error 
+      end  
+      if (!params[:date_coverage_1_month].eql?("--"))
+        mm1 =  params[:date_coverage_1_month].to_i   
+      else
+        mm1 = 1    
```

Comment:
Ruby's Date library does assume the first month and the first day when not supplied with one.  In our use case not specified day and month will set the precision of the date.  The edtf library has provisions for this.  Setting the parameter when one was not supplied masks the given precision.